File: backend/app/Services/AI/ProviderRegistry.php
<?php

namespace App\Services\AI;

use App\Services\AI\Adapters\NVIDIAAdapter;
use App\Services\AI\Adapters\OllamaAdapter;
use App\Services\AI\Adapters\GroqAdapter;
use App\Services\AI\Adapters\CerebrasAdapter;
use App\Services\AI\Adapters\ProviderAdapterInterface;

class ProviderRegistry
{
    protected array $adapters = [];
    protected array $order = [];

    public function __construct()
    {
        $this->loadConfiguration();
    }

    protected function loadConfiguration(): void
    {
        $orderEnv = env('AI_PROVIDER_ORDER', 'nvidia,ollama,groq,cerebras');
        $this->order = array_filter(array_map('trim', explode(',', $orderEnv)));

        foreach ($this->order as $providerKey) {
            $prefix = 'AI_' . strtoupper($providerKey) . '_';
            
            if (env($prefix . 'ENABLED', false)) {
                $config = [
                    'id' => $providerKey,
                    'base_url' => env($prefix . 'BASE_URL'),
                    'api_key' => env($prefix . 'API_KEY'),
                    'model' => env($prefix . 'MODEL'),
                    'vision_model' => env($prefix . 'VISION_MODEL'),
                ];

                $adapter = $this->createAdapter($providerKey, $config);
                if ($adapter) {
                    $this->adapters[$providerKey] = $adapter;
                }
            }
        }
    }

    protected function createAdapter(string $providerKey, array $config): ?ProviderAdapterInterface
    {
        return match (strtolower($providerKey)) {
            'nvidia' => new NVIDIAAdapter($config),
            'ollama' => new OllamaAdapter($config),
            'groq' => new GroqAdapter($config),
            'cerebras' => new CerebrasAdapter($config),
            default => null,
        };
    }

    public function getAdapters(): array
    {
        return $this->adapters;
    }

    public function getAdapter(string $id): ?ProviderAdapterInterface
    {
        return $this->adapters[$id] ?? null;
    }

    public function getOrder(): array
    {
        return $this->order;
    }
}

File: worker/new/services/ai/provider_registry.py
import os
from typing import Dict, List, Optional
from dotenv import load_dotenv

load_dotenv()

class ProviderRegistry:
    def __init__(self):
        self.providers: Dict[str, dict] = {}
        self.order: List[str] = []
        self._load_configuration()

    def _load_configuration(self):
        order_env = os.getenv('AI_PROVIDER_ORDER', 'nvidia,ollama,groq,cerebras,mock')
        self.order = [p.strip() for p in order_env.split(',') if p.strip()]

        for provider_key in self.order:
            prefix = f"AI_{provider_key.upper()}_"
            
            if os.getenv(f"{prefix}ENABLED", "false").lower() == "true":
                self.providers[provider_key] = {
                    'id': provider_key,
                    'base_url': os.getenv(f"{prefix}BASE_URL"),
                    'api_key': os.getenv(f"{prefix}API_KEY", ""),
                    'model': os.getenv(f"{prefix}MODEL"),
                    'vision_model': os.getenv(f"{prefix}VISION_MODEL"),
                }

    def get_providers(self) -> Dict[str, dict]:
        return self.providers

    def get_provider(self, provider_id: str) -> Optional[dict]:
        return self.providers.get(provider_id)

    def get_order(self) -> List[str]:
        return self.order

# Singleton instance
registry = ProviderRegistry()
